### PR TITLE
add sonarqube plugin

### PR DESCRIPTION
**Pull request by roganw**
The plugin of Drone CI to integrate with SonarQube (previously called Sonar), which is an open source code quality management platform.

This plugin will automatically post the code analysis report to SonarQube immediately after the execution of the associated pipeline finished.

### CODE REVIEW

**Comment by tboerger** [comment]
Thank you for this plugin, but could you please follow the style of other plugins like https://github.com/drone/drone-plugin-index/blob/master/content/drone-plugins/drone-codecov/index.m?

**Comment by roganw** [comment]
@tboerger Done. Please review it again, thx.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ content/**.gen.*
 *.txt
 cmd
 server.go
+.history
```
Comment (appleboy) [inline .gitignore]: Why need this?

Comment (roganw) [reply .gitignore]: It's generated by VS code, including the change history of code.

Comment (appleboy) [reply .gitignore]: Please revert this change.

**Pull request merged**